biz.aQute.launchpad/test/aQute/launchpad/LaunchpadTest.java

```diff
@@ -464,7 +464,9 @@ public void testLaunchpadStressByCreatingLotsOfFrameworksInDifferentThreads() th
 					}
 				});
 		}
-		semaphore.acquire(n);
+		System.out.println("Waiting for the threads to finish");
+		boolean success = semaphore.tryAcquire(n, 60, TimeUnit.SECONDS);
+		assertTrue(success);
```

Comment:
I think you want to softly assert this so that you still close the frameworks.

Reply:
Well, the test is theoretically not supposed to be able to timeout, if the time out happens you're in deeper shit than a missed resource?

But the softly asserts sounds nice

Reply:
Theoretically, hopefully, none of the tests should fail... :-)